ts/Stock/StockToolsBindings.ts

```diff
@@ -2047,41 +2047,40 @@ var stockToolsBindings: Record<string, Highcharts.NavigationBindingsOptionsObjec
             button: HTMLDOMElement
         ): void {
             var chart = this.chart,
-                series = chart.series[0],
-                options = series.options,
-                lastVisiblePrice = (
-                    options.lastVisiblePrice &&
-                    options.lastVisiblePrice.enabled
-                ),
-                lastPrice = options.lastPrice && options.lastPrice.enabled,
-                gui: Highcharts.Toolbar = chart.stockTools as any,
-                iconsURL = gui.getIconsURL();
-
-            if (gui && gui.guiEnabled) {
-                if (lastPrice) {
-                    (button.firstChild as any).style['background-image'] =
-                        'url("' + iconsURL +
-                        'current-price-show.svg")';
+                series = chart.series,
+                gui: Highcharts.Toolbar = chart.stockTools as any;
+
+            if (gui && gui.guiEnabled && chart.stockTools) {
+                // Reset all and disable both indicators
+                if (bindingsUtils.isPriceIndicatorEnabled(chart.series)) {
+                    series.forEach(function (series): void {
+                        series.update({
+                            lastPrice: {
+                                enabled: false
+                            },
+                            lastVisiblePrice: {
+                                enabled: false
+                            }
+                        }, false);
+                    });
                 } else {
-                    (button.firstChild as any).style['background-image'] =
-                        'url("' + iconsURL +
-                        'current-price-hide.svg")';
+                    series.forEach(function (series): void {
+                        series.update({
+                            lastPrice: {
+                                enabled: true
+                            },
+                            lastVisiblePrice: {
+                                enabled: true,
+                                label: {
+                                    enabled: true
+                                }
+                            }
+                        }, false);
```

Comment:
Q: Would that make sense to store `bindingsUtils.isPriceIndicatorEnabled(chart.series)` in a local variable? Then we could simplify this long if-else to:

```js
const priceIndicatorEnabled = bindingsUtils.isPriceIndicatorEnabled(chart.series);

if (gui && gui.guiEnabled && chart.stockTools) {
  series.forEach(function (series): void {
    series.update({
      lastPrice: { enabled: !priceIndicatorEnabled },
      lastVisiblePrice: { enabled: !priceIndicatorEnabled, label: { enabled: true } }
    }, false);
  });
}
```